ember-client/src/main/scala/org/http4s/ember/client/EmberClientBuilder.scala

```diff
@@ -83,6 +85,11 @@ final class EmberClientBuilder[F[_]: Concurrent: Timer: ContextShift] private (
   def withAdditionalSocketOptions(additionalSocketOptions: List[SocketOptionMapping[_]]) =
     copy(additionalSocketOptions = additionalSocketOptions)
 
+  def withUserAgent(userAgent: `User-Agent`) =
+    copy(userAgent = userAgent.some)
+  def withoutUserAgent =
+    copy(userAgent = None)
```

Comment:
This feature feels like a potential new middleware, though I think it's also good here, since a reasonable default depends on the backend.